File: backend/app/models/ai_assistant.py
from pydantic import BaseModel, Field
from typing import Optional


class ChatRequest(BaseModel):
    """
    Request para el chat del asistente.
    """
    message: str = Field(..., min_length=1, description="Mensaje del usuario")
    selected_mode: Optional[str] = Field(
        None,
        description="Modo/herramienta seleccionada (opcional)"
    )
    collection_id: Optional[str] = Field(
        None,
        description="ID de coleccion para acotar el contexto del asistente (opcional)",
    )
    runtime_mode: Optional[str] = Field(
        None,
        description="Modo runtime del asistente: online u offline (opcional)",
    )
    web_provider: Optional[str] = Field(
        None,
        description="Proveedor de busqueda web (opcional): duckduckgo o hackernews",
    )

File: frontend/src/api/aiAssistant.js
import { apiFetch } from './client.js'

export const aiAssistantAPI = {
  chat: async (message, selected_mode, collection_id = null, runtime_mode = null, web_provider = null) => (
    apiFetch('/ai-assistant/chat', {
      method: 'POST',
      body: JSON.stringify({
        message,
        selected_mode: selected_mode || null,
        collection_id: collection_id || null,
        runtime_mode: runtime_mode || null,
        web_provider: web_provider || null,
      }),
    })
  ),
}
